File: 151003/Saydalikhujaev/publisher/src/main/java/org/example/publisher/impl/creator/mapper/CreatorMapper.java
package org.example.publisher.impl.creator.mapper;

import org.example.publisher.impl.creator.Creator;
import org.example.publisher.impl.creator.dto.CreatorRequestTo;
import org.example.publisher.impl.creator.dto.CreatorResponseTo;

import java.util.List;

public interface CreatorMapper {

    CreatorRequestTo creatorToRequestTo(Creator creator);

    List<CreatorRequestTo> creatorToRequestTo(Iterable<Creator> creators);

    Creator dtoToEntity(CreatorRequestTo creatorRequestTo);

    List<Creator> dtoToEntity(Iterable<CreatorRequestTo> creatorRequestTos);

    CreatorResponseTo creatorToResponseTo(Creator creator);

    List<CreatorResponseTo> creatorToResponseTo(Iterable<Creator> creators);
    
    
}

File: 151001/Hlebanova/LR1/LR1_REST/Mappers/CreatorMapper.cs
using Riok.Mapperly.Abstractions;
using LR1.Dto.Request.CreateTo;
using LR1.Dto.Request.UpdateTo;
using LR1.Dto.Response;
using LR1.Models;

namespace LR1.Mappers;

[Mapper]
public static partial class CreatorMapper
{
    public static partial Creator Map(UpdateCreatorRequestTo updateCreatorRequestTo);
    public static partial Creator Map(CreateCreatorRequestTo createCreatorRequestTo);
    public static partial CreatorResponseTo Map(Creator creator);
    public static partial IEnumerable<CreatorResponseTo> Map(IEnumerable<Creator> creators);

    public static partial IEnumerable<Creator> Map(
        IEnumerable<UpdateCreatorRequestTo> creatorRequestTos);
}